activesupport/lib/active_support/core_ext/thread/backtrace/location.rb

```diff
@@ -9,4 +9,7 @@ def spot(ex)
     def spot(ex)
     end
   end
+  def thread_backtrace_location
```

Comment:
```suggestion

    def thread_backtrace_location
```

I don't like the idea of adding a core_ext for this.

I'd rather use `is_a?` in the caller. Polymorphism is nicer indeed, but not at the cost of reopening a core class.

Reply:
Thanks @byroot! 

I can change the logic to use `is_a?`, but I wanted to keep in line with the spirit of https://github.com/rails/rails/pull/46171 where `Thread::Backtrace::Location` has been monkey patched. 

My fear is that if I change this code to use `is_a?`, then this is not consistent with the rest of the code, which is instead relying on `Thread::Backtrace::Location` being monkey patched.

I am happy to use `is_a?` but I wanted to hear your thoughts.

Reply:
Yeah, I disagree with how @tenderlove implemented it, I think it would have been possible to not add that `Location#spot` method.

Reply:
> Yeah, I disagree with how @tenderlove implemented it, I think it would have been possible to not add that `Location#spot` method.

If you want to refactor, I'm fine with it! 😂

One alternative would be to implement `itself` on `BacktraceLocationProxy`.  Then we wouldn't need to add another method to `Thread::Backtrace::Location`

TBH, I think this `def spot(ex)` monkey patch should actually be part of ErrorHighlight.  ErrorHighlight only works with BacktraceLocation objects, so the double dispatch / factory pattern makes sense to me.

Reply:
> If you want to refactor, I'm fine with it! 😂

I know :)

I'll see if get some time to avoid that patch, in the meantime I think we should mark many of this stuff as private API: https://edgeapi.rubyonrails.org/classes/Thread/Backtrace/Location.html#method-i-spot

Reply:
Thanks guys!

I have made the following changes to `exception_wrapper.rb`:

- Before calling `spot` on the template object, check if we have a real Thread::Backtrace::Location object to prevent calling MRI with the wrong object;
- When building the SourceMapLocation object, ensure that we are passing a Thread::Backtrace::Location object and not a BacktraceLocationProxy object